src/components/plots/barPlot.ts

```diff
@@ -43,6 +43,63 @@ export module Plot {
       this._baseline = this._renderArea.append("line").classed("baseline", true);
     }
 
+    protected _onDatasetUpdate() {
+      super._onDatasetUpdate();
+      if (this._isAnchored) {
+        this._updateBarExtent();
+      }
+    }
+
+    public detach() {
+      super.detach();
+      this._datasetKeysInOrder.forEach((key) => {
+        var scaleKey = this.getID().toString() + "_" + key;
+        this._xScale._removeExtent(this.getID().toString(), "bar-extent");
```

Comment:
I wonder if `getID()` should just always return strings. There's no point in doing math on an ID...?